src/core/auth/service/auth.service.ts

```diff
@@ -0,0 +1,200 @@
+import {
+  BadRequestException,
+  HttpException,
+  HttpStatus,
+  Injectable,
+  Logger,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { LoginDto } from '../dto/login.dto';
+import { RegisterDto } from '../dto/register.dto';
+import { UsersService } from '../../users/service/users.service';
+import { JwtService } from '@nestjs/jwt';
+import * as bcryptjs from 'bcryptjs';
+import { UserRole } from '../../../constants';
+import { InjectRepository } from '@nestjs/typeorm';
+import { EmailVerificationEntity } from '../entities/emailverification.entity';
+import { Repository } from 'typeorm';
+import { JwtPayload } from '../interface/jwt-payload.interface';
+import 'dotenv/config';
+import { JwtPayloadService } from '../../../common/service/jwt.payload.service';
+import { transporter } from '../../../constants';
+import * as handlebars from 'handlebars';
+import * as fs from 'fs';
+import * as path from 'path';
+import { USER_NOT_FOUND } from 'src/core/users/messages';
+import {
+  INVALID_TOKEN_EMAIL_MESSAGE,
+  SUCCESSFULY_SEND_EMAIL_VERIFICATION_MESSAGE,
+  USER_ALREADY_EXISTS_MESSAGE,
+} from '../message';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    @InjectRepository(EmailVerificationEntity)
+    private readonly emailVerification: Repository<EmailVerificationEntity>,
+
+    private readonly usersService: UsersService,
+
+    private readonly jwtService: JwtService,
+
+    private readonly jwtPayloadService: JwtPayloadService,
+  ) {}
+
+  async register({ email, password, name, lastname }: RegisterDto) {
+    const user = await this.usersService.findOneByEmail(email);
+
+    if (user) {
+      throw new BadRequestException(USER_ALREADY_EXISTS_MESSAGE);
+    }
+
+    await this.usersService.create({
+      email,
+      password: await bcryptjs.hash(password, 10),
+      name,
+      lastname,
+      role: UserRole.GRADUATE,
+    });
+
+    return {
+      email,
+      name,
+    };
+  }
+
+  async login({ email, password }: LoginDto) {
+    const user = await this.usersService.findByEmailWithPassword(email);
+    if (!user) {
+      throw new UnauthorizedException('Correo incorrecto');
+    }
+
+    const isPasswordValid = await bcryptjs.compare(password, user.password);
+
+    if (!user.verified) {
+      throw new UnauthorizedException('Correo no verificado');
+    }
+
+    if (!isPasswordValid) {
+      throw new UnauthorizedException('Contraseña incorrecta');
+    }
+
+    const payload = { email: user.email, id: user.id, role: user.role };
+    const token = await this.jwtService.signAsync(payload, {
+      secret: process.env.JWT_SECRET,
```

Comment:
Usar constantes intermedias, no directamente el `process.env.JWT_SECRET`
```suggestion
      secret: JWT_SECRET,
```